### DISCUSSION TITLE
Mask a point cloud by a second one

### DISCUSSION THREAD

**Discussion by CH-NWA** [r-lidar/lidR#645] [Ideas]
Dear lidR-Team,
is there a direct function with which point clouds can be masked by others? My feature request works like filter_duplicates but in another way. I compare two point clouds and the function classifies which points are identical or not, based on their exact position or within a given spatial resolution/voxel size.
My first workaround for this is the use of voxel_metrics: 
```
las = readLAS(files=c(las1,las2))
df = voxel_metrics(las, ~list(N = length(Z)), 0.00001) 
if(df$N >=2) => points are identical 
```
Compared to voxel_metrics a direct function which keeps the LAS structure intact would be very nice :)

**Comment by Jean-Romain** [comment]
You can take the code of `filter_duplicates()` and do something else with the result

```r
  dup_xyz <- duplicated(las@data, by = c("X", "Y", "Z"))
```

**Comment by CH-NWA** [comment]
Yes, I already checked this function in filters.R, but did not try this function directly in my script. I will give it a try.
